### PR TITLE
Refactor state

### PR DESCRIPTION
**Pull request by xwatkins**
Break up form state into smaller state items.

### CODE REVIEW

File: src/tools/blast/components/BlastForm.tsx [hunk 1]
```diff
@@ -67,31 +67,32 @@ const getAutoMatrixFor = (sequence: string): FormParameters['matrix'] => {
...
+}> = ({ formValue, updateFormValue }) => {
+  if (!formValue) {
     return null;
   }
+  const label = BlastFields[formValue.fieldName as keyof typeof BlastFields];
   return (
     <section className="blast-form-section__item">
-      <label htmlFor={type}>
-        {type}
+      <label htmlFor={label}>
```
Comment (aurel-l) [inline src/tools/blast/components/BlastForm.tsx:79]: The label here is wrapping the corresponding select element, so no need to had extra ids and for attributes to link them.

File: src/tools/blast/components/BlastForm.tsx [hunk 2]
```diff
@@ -108,83 +109,98 @@ const BlastForm = () => {
   const dispatch = useDispatch();
   const history = useHistory();
 
+  const getInitialValues = () => {
```
Comment (aurel-l) [inline src/tools/blast/components/BlastForm.tsx:112]: this probably should be wrapped in a react `useMemo` function as it's quite heavy logic on each render for something used only once on mount

File: src/tools/blast/config/BlastFormData.ts [hunk 3]
```diff
@@ -58,7 +58,7 @@ export default Object.freeze({
     selected: '',
   }),
   [BlastFields.targetDb]: Object.freeze({
-    fieldName: 'database',
+    fieldName: 'targetDb',
```
Comment (aurel-l) [inline src/tools/blast/config/BlastFormData.ts:61]: This should probably have had been changed the other way around, as 'database' is what is used in the internal job parameters (`FormParameters`) and on the server (`ServerParameters`)

File: src/tools/blast/components/BlastForm.tsx (hunk 1 above)
Comment (xwatkins) [reply src/tools/blast/components/BlastForm.tsx:80]: Done

File: src/tools/blast/components/BlastForm.tsx (hunk 2 above)
Comment (xwatkins) [reply src/tools/blast/components/BlastForm.tsx]: Done

File: src/tools/blast/config/BlastFormData.ts (hunk 3 above)
Comment (xwatkins) [reply src/tools/blast/config/BlastFormData.ts]: Done

**Pull request merged**